### DISCUSSION TITLE
Old ReplicaSets are not being removed.

### DISCUSSION THREAD

**Discussion by axelluguercio** [argoproj/argo-rollouts#3568] [Q&A]
I am noticing that when I start a rollout, the analysis ends successfully, the canary pod becomes stable, but the pod from the previous revision is still running.

Version: v.1.6.6

Rollout spec:

apiVersion: argoproj.io/v1alpha1
kind: Rollout
metadata:
  annotations:
    meta.helm.sh/release-name: xxxxxx
    meta.helm.sh/release-namespace: xxxxxx
    rollout.argoproj.io/revision: '2'
    rollout.argoproj.io/workload-generation: '3'
  creationTimestamp: '2024-05-10T19:15:26Z'
  generation: 2
  labels:
    app.kubernetes.io/managed-by: Helm
    chart: canary
    release: xxxxxx
status:
  HPAReplicas: 2
  availableReplicas: 2
  blueGreen: {}
  canary:
    weights:
      canary:
        podTemplateHash: 7fcb899664
        weight: 0
      stable:
        podTemplateHash: 7fcb899664
        weight: 100
  conditions:
    - lastTransitionTime: '2024-05-10T19:15:33Z'
      lastUpdateTime: '2024-05-10T19:15:33Z'
      message: Rollout has minimum availability
      reason: AvailableReason
      status: 'True'
      type: Available
    - lastTransitionTime: '2024-05-10T19:20:29Z'
      lastUpdateTime: '2024-05-10T19:20:29Z'
      message: Rollout is not healthy
      reason: RolloutHealthy
      status: 'False'
      type: Healthy
    - lastTransitionTime: '2024-05-10T19:26:34Z'
      lastUpdateTime: '2024-05-10T19:26:34Z'
      message: Rollout is paused
      reason: RolloutPaused
      status: 'False'
      type: Paused
    - lastTransitionTime: '2024-05-10T19:26:34Z'
      lastUpdateTime: '2024-05-10T19:26:34Z'
      message: RolloutCompleted
      reason: RolloutCompleted
      status: 'True'
      type: Completed
    - lastTransitionTime: '2024-05-10T19:36:34Z'
      lastUpdateTime: '2024-05-10T19:36:34Z'
      message: >-
        ReplicaSet "deployment-xxxxx-7fcb899664" has timed out
        progressing.
      reason: ProgressDeadlineExceeded
      status: 'False'
      type: Progressing
  currentPodHash: 7fcb899664
  currentStepHash: 5cfd46867b
  currentStepIndex: 9
  message: >-
    ProgressDeadlineExceeded: ReplicaSet
    "deployment-xxxx-7fcb899664" has timed out progressing.
  observedGeneration: '2'
  phase: Degraded
  readyReplicas: 2
  replicas: 2
  selector: app=xxxxxx
  stableRS: 7fcb899664
  updatedReplicas: 1
  workloadObservedGeneration: '3'
spec:
  analysis:
    successfulRunHistoryLimit: 5
    unsuccessfulRunHistoryLimit: 5
  replicas: 1
  revisionHistoryLimit: 5
  rollbackWindow:
    revisions: 5
  selector:
    matchLabels:
      app: xxxxxxx
  strategy:
    canary:
      analysis:
        analysisRunMetadata: {}
        args:
          - name: service
            value: xxxxx
          - name: interval
            value: 6m
        templates:
          - clusterScope: true
            templateName: error-rate
          - clusterScope: true
            templateName: latency
      canaryMetadata:
        labels:
          role: canary
          version: canary
      scaleDownDelaySeconds: 0
      stableMetadata:
        labels:
          role: stable
          version: stable
      steps:
        - setWeight: 20
        - pause:
            duration: 1m
        - setWeight: 40
        - pause:
            duration: 1m
        - setWeight: 60
        - pause:
            duration: 2m
        - setWeight: 80
        - pause:
            duration: 2m
        - setWeight: 100
      trafficRouting:
        istio:
          destinationRule:
            canarySubsetName: canary
            name: destinationrule-xxxx
            stableSubsetName: stable
          virtualService:
            name: virtualservice-xxxxx
  workloadRef:
    apiVersion: apps/v1
    kind: Deployment
    name: deployment-xxxxx
    scaleDown: onsuccess

Name:            deployment-xxxxxxxxx
Namespace:       xxxxxxxxxxx
Status:          ✖ Degraded
Message:         ProgressDeadlineExceeded: ReplicaSet "deployment-xxxxxxxxx-7fcb899664" has timed out progressing.
Strategy:        Canary
  Step:          9/9
  SetWeight:     100
  ActualWeight:  100
Images:          xxxxxxxxx/xxxxxxxx/xxxxxxxx:1.0.0-2024.05.10-12.49.09 (stable)
                 xxxxxxxxx/xxxxxxxxx/xxxxxxxxxx:1.0.0-2024.05.10-14.04.44
Replicas:
  Desired:       1
  Current:       2
  Updated:       1
  Ready:         2
  Available:     2

NAME                                                        KIND         STATUS        AGE  INFO
⟳ deployment-xxxxxxx                            Rollout      ✖ Degraded    62m
├──# revision:2
│  ├──⧉ deployment-xxxxxx-7fcb899664           ReplicaSet   ✔ Healthy     57m  stable
│  │  └──□ deployment-xxxxx-7fcb899664-5hsdq  Pod          ✔ Running     57m  ready:2/2
│  └──α deployment-xxxxxxxx-7fcb899664-2         AnalysisRun  ✔ Successful  57m  ✔ 14
└──# revision:1
   └──⧉ deployment-xxxxx-6d86c795f6           ReplicaSet   ✔ Healthy     62m
      └──□ deployment-bbe-xxxxxx-6d86c795f6-m2ffl  Pod          ✔ Running     62m  ready:2/2

Please any light on this?

**Comment by kostis-codefresh** [comment]
Can you post the original yaml with correct formatting please? Not the live object from the Kubernetes cluster.
